trpcultivate_phenotypes/src/Plugin/Validators/ProjectGenusMatch.php

```diff
@@ -154,4 +177,107 @@ public function validateMetadata(array $form_values) {
     ];
   }
 
+  /**
+   * Process failed validation from ProjectGenusMatch into a render array.
+   *
+   * @param array $failure
+   *   An associative array that was returned by the ProjectGenusMatch validator
+   *   in the event of failed validation. It contains the following keys:
+   *   - 'case': a developer-focused string describing the case checked.
+   *   - 'valid': FALSE to indicate that validation failed.
+   *   - 'failedItems': an array of items that failed with the following keys.
+   *     - 'project_provided': The name of the project provided.
+   *     - 'genus_provided': The name of the genus provided.
+   * @param array $tokens
+   *   [OPTIONAL] An array of values to use for token replacement.
+   *   The following tokens can be specfied as keys, with value as the
+   *   replacement value for the token. These apply to all failure cases.
+   *   - 'project': replaces the word "project".
+   *   - 'contact-admin': replaces the phrase "contact your administrator".
+   *   The following token keys will substitute the entire existing case message
+   *   with the value of that token.
+   *   - 'case-message-1': "Project does not exist"
+   *   - 'case-message-2': "Project has no genus set and could not compare with
+   *     the genus provided"
+   *   - 'case-message-3': "Genus does not match a genus set to the project".
```

Comment:
Since these are actually in the default_tokens variable, we could add an `@see` to reference that. That way if these change we don't end up with the docblock not matching the values. Then we can define these tokens more free-text like. For example, 
`- 'case-message-1': the message to provide to the user when the project entered does not exist.`.

Reply:
This is a great suggestion - I also applied it to the 'project' and 'contact-admin' tokens above this. This is also addressed in commit https://github.com/TripalCultivate/TripalCultivate-Phenotypes/pull/159/commits/c4583ed2de1ff495adc9723f431e59f7c87a3cb9

Reply:
Yes, love these updates!